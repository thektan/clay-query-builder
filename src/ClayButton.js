import React from 'react';
import {PropTypes} from 'prop-types';

class ClayButton extends React.Component {
    render() {
        const {label, type} = this.props;

        return (
            <button className="btn btn-secondary" {...this.props}>{label}</button>
        );
    }
}

ClayButton.propTypes = {
    label: PropTypes.string,
    type: PropTypes.string
}

export default ClayButton;