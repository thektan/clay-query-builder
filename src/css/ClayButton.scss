.icon {
	margin: 0;
}

.text-icon {
	margin-left: 8px;
}