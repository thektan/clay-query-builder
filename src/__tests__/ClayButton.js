import React from 'react';
import ClayButton from '../ClayButton';
import renderer from 'react-test-renderer';

describe('ClayButton', () => {
	it('should render', () => {
		const component = renderer.create(<ClayButton label="test" />);

		let tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});