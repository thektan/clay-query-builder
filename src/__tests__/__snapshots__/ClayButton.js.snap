// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayButton should render 1`] = `
<button
  className="btn btn-secondary "
>
  test
</button>
`;
